src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -791,11 +808,12 @@ public TypeHandle GetArrayElementTypeHandle()
     }
 
     // Subset of src\vm\methodtable.h
-    [StructLayout(LayoutKind.Explicit)]
+    [StructLayout(LayoutKind.Sequential)]
     internal unsafe struct MethodTableAuxiliaryData
     {
-        [FieldOffset(0)]
         private uint Flags;
+        private nint m_pLoaderModule;
```

Comment:
```suggestion
        private void* m_pLoaderModule;
```
This is actual unmanaged pointer, so `void*` may be more appropriate.